Additional EH tests

russellhadley opened the pull request:
Adding some more selected EH tests.  At this point I'm still not including
particular tests that were repros for MSVC compiler bugs since more work is
needed to tell if these are useful.

russellhadley committed:
Additional EH tests

Adding some more selected EH tests.  At this point I'm still not including
particular tests that were repros for MSVC compiler bugs since more work is
needed to tell if these are useful.

russellhadley commented:
@majnemer - PTAL

majnemer commented on eh/ehthrow.cxx:
```diff
@@ -0,0 +1,547 @@
+    throw A();
+}
+
+void main()
```
This will not compile with clang, can you make the return type `int`?

majnemer commented on eh/nesttry.cxx:
```diff
@@ -0,0 +1,38 @@
+    throw 1;
+}
+
+void main()
```
Ditto here.

majnemer commented on eh/noreturn.cpp:
```diff
@@ -0,0 +1,54 @@
+    bar(a1);
+}
+
+void main()
```
Ditto here.

majnemer commented on eh/recursive_throw.cpp:
```diff
@@ -0,0 +1,77 @@
+    }
+}
+
+void main(void)
```
Ditto here.

majnemer commented on eh/noreturn.cpp:
```diff
@@ -0,0 +1,54 @@
+    throw 1;
+}
+
+void foo(A& a, int i)
```
The `a` parameter needs to be a `const` l-value reference: `const A& a`.

majnemer commented on eh/terminate.cpp:
```diff
@@ -0,0 +1,123 @@
+        if (FGenSEHException)
+        {
+            printf("generating access violation (SEH)\n");
+            *(char*)(0) = 0; // Uh, EHa, don't you think?
```
This should be a volatile store: `*(volatile char*)(0) = 0;`

majnemer commented:
Looks good, found some trivial issues.

russellhadley committed:
Update with fixes for clang build.

Change main() returns to 'int' and modify a passed parameter reference to const.

russellhadley commented:
@majnemer made the changes you suggested.  Merging now.
